force-app/src/classes/MethodSpy.cls

```diff
@@ -324,33 +352,31 @@ global class MethodSpy {
       return this.generalBehavior != null;
     }
 
-    public Object pollConfiguredBehavior() {
-      final ConfiguredBehavior behaviorToApply = this.hasConfiguredOnceBehavior() ? this.onceBehaviors.remove(0) : this.generalBehavior;
-      return behaviorToApply?.apply();
+    public Object pollSpyBehavior(final List<Object> args) {
+      final SpyBehavior behaviorToApply = this.hasConfiguredOnceBehavior() ? this.onceBehaviors.remove(0) : this.generalBehavior;
+      return behaviorToApply?.execute(args);
     }
 
     public override String toString() {
       final List<String> oncesString = new List<String>();
-      for (ConfiguredBehavior configuredBehaviorOnce : this.onceBehaviors) {
+      for (SpyBehavior configuredBehaviorOnce : this.onceBehaviors) {
         oncesString.add(configuredBehaviorOnce.toString());
       }
       return String.join(oncesString, '') + (this.generalBehavior != null ? this.generalBehavior.toString() : '');
     }
   }
 
-  private interface ConfiguredBehavior {
-    Object apply();
+  global interface SpyBehavior {
+    Object execute(List<Object> params);
   }
```

Comment:
note: in the future, if we need to have new methods for private usage only (no user impact), we'll have to re-introduce something like `ConfiguredBehavior` interface and we can either link them with inheritance or use the private one for all private code directly as soon at it has at least the very same contract as `SpyBehavior` interface